Assets/Character/Scripts/PlayerState/PlayerState.cs

```diff
@@ -17,11 +19,18 @@ public PlayerState(Animator animator)
         _playerInput = null;
         _playerInputHandling = true;
         _selfTransition = false;
+        _stamina = _player.GetComponent<Stamina>();
+        _staminaRechargeTime = Constants.PLAYER_STAMINA_CHARGE_TIME;
+        
+        // Stamina and health are 0 by default
+        _stamina.reset(); 

```

Comment:
reset shouldn't be required after initialization. make sure Stamina does this when initialized.